packages/edit-navigation/src/filters/add-menu-name-editor.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * WordPress dependencies
+ */
+/**
+ * Internal dependencies
+ */
+import { NameEditor, useNavigationEditorMenu } from '../components/name-editor';
+import { addFilter } from '@wordpress/hooks';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+const addMenuNameEditor = createHigherOrderComponent(
+	( BlockEdit ) => ( props ) => {
+		if ( props.name !== 'core/navigation' ) {
+			return <BlockEdit { ...props } />;
+		}
+		const { saveMenu, menuName } = useNavigationEditorMenu();
+		const NavigationBlockEdit = () =>
+			props.name !== 'core/navigation' ? (
+				<BlockEdit { ...props } />
+			) : (
+				<BlockEdit
+					{ ...props }
+					saveMenu={ saveMenu }
+					menuName={ menuName }
+				/>
+			);
```

Comment:
I don't think this inline component is required because the `if` statement on line 13 already does this check.

Should be able to pass the extra props straight to `<BlockEdit />` in the return statement below this.

Reply:
fixed